NYTPhotoViewer/NYTPhotosViewController.h

```diff
@@ -85,40 +97,39 @@ extern NSString * const NYTPhotosViewControllerDidDismissNotification;
+- (instancetype)initWithDataSource:(id <NYTPhotoViewerDataSource>)dataSource;
 
 /**
- *  A convenience initializer that calls `initWithPhotos:initialPhoto:delegate:`, passing `nil` as the `delegate` argument.
+ *  Initializes a `PhotosViewController` with the given data source and delegate, initially displaying the photo at the given index in the data source.
  *
- *  @param photos An array of objects conforming to the `NYTPhoto` protocol.
- *  @param initialPhoto The photo to display initially. Must be contained within the `photos` array. If `nil` or not within the `photos` array, the first photo within the `photos` array will be displayed.
+ *  @param dataSource        The data source underlying this photo viewer.
+ *  @param initialPhotoIndex The photo to display initially. If outside the bounds of the data source, the first photo from the data source will be displayed.
+ *  @param delegate          The delegate for this `NYTPhotosViewController`.
  *
- *  @return A fully initialized object.
+ *  @return A fully initialized `PhotosViewController` instance.
  */
-- (instancetype)initWithPhotos:(NSArray <id <NYTPhoto>> * _Nullable)photos initialPhoto:(id <NYTPhoto> _Nullable)initialPhoto;
+- (instancetype)initWithDataSource:(id <NYTPhotoViewerDataSource>)dataSource initialPhotoIndex:(NSInteger)initialPhotoIndex delegate:(nullable id <NYTPhotosViewControllerDelegate>)delegate;
 
 /**
- *  The designated initializer that stores the array of objects conforming to the `NYTPhoto` protocol for display, along with specifying an initial photo for display.
+ *  Initializes a `PhotosViewController` with the given data source and delegate, initially displaying the given photo.
  *
- *  @param photos An array of objects conforming to the `NYTPhoto` protocol.
- *  @param initialPhoto The photo to display initially. Must be contained within the `photos` array. If `nil` or not within the `photos` array, the first photo within the `photos` array will be displayed.
- *  @param delegate The delegate for this `NYTPhotosViewController`.
+ *  @param dataSource   The data source underlying this photo viewer.
+ *  @param initialPhoto The photo to display initially. Must be a member of the data source. If `nil` or not a member of the data source, the first photo from the data source will be displayed.
+ *  @param delegate     The delegate for this `NYTPhotosViewController`.
  *
- *  @return A fully initialized object.
+ *  @return A fully initialized `PhotosViewController` instance.
  */
-- (instancetype)initWithPhotos:(NSArray <id <NYTPhoto>> * _Nullable)photos initialPhoto:(id <NYTPhoto> _Nullable)initialPhoto delegate:(nullable id <NYTPhotosViewControllerDelegate>)delegate NS_DESIGNATED_INITIALIZER;
+- (instancetype)initWithDataSource:(id <NYTPhotoViewerDataSource>)dataSource initialPhoto:(id <NYTPhoto> _Nullable)initialPhoto delegate:(nullable id <NYTPhotosViewControllerDelegate>)delegate NS_DESIGNATED_INITIALIZER;
```

Comment:
Aren't _Nullable and nullable equivalent? Is there a reason not to be consistent? I guess this was already here but it might be nice to make the usage consistent and settle on one style.